core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/RateAppCounter.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.utils.dialog
+
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import javax.inject.Inject
+
+class RateAppCounter {
+
+  @Inject
+  lateinit var sharedPreferencesUtil: SharedPreferenceUtil
+  var noThanksState: Boolean = sharedPreferencesUtil.noThanksState
```

Comment:
I very much doubt this still functions. The preferences will never be set. Additionally this creates a whole new preference  in the default file so our existing rateApp preferences (in their very weird, own sharedPreference files) have been abandoned and I would expect this PR to maintain data integrity even though the implementation is a good bit mental

Reply:
```
    visitCounter = context.getSharedPreferences(MASTER_NAME, 0);
    visitCounter = context.getSharedPreferences(NOTHANKS_CLICKED, 0);
```
blugh, apparently the "NOTHANKS_CLICKED" preferences file is the only one that is used. So unpleasant and misleading

Reply:
@macgills I am so sorry. I don't know about Dagger. I will learn as soon as possible. Should I close the pull request?

Reply:
You can change this code, its fine. You just don't want SharedPrefUtil at all because that is not what the original code did. I made a post in the skillshare channel on slack about Dagger, you can read that if you want some clarity but Dagger isn't necessary to convert this

I'll just translate this kotlin code
```
var noThanksState: Boolean = sharedPreferencesUtil.noThanksState
```
to java
```
private boolean noThanksState = sharedPreferencesUtil.getNoThanksState()

public boolean getNoThanksState(){
return noThanksState;
}

public void setNoThanksState(boolean aBoolean){
noThanksState = aBoolean;
```
and hopefully that should clear up why this does not set the preferences